logpolicy/logpolicy.go

```diff
@@ -573,6 +540,49 @@ func New(collection string) *Policy {
 	}
 }
 
+// LoadOrCreateConfig will load the logpolicy.Config from the specified path.
+// If it does not exist, a new config will be created.  If provided collection
+// does not match the loaded config, it will be overwritten and a new logID
+// generated.  Changes are saved back to cfgPath.
+func LoadOrCreateConfig(cfgPath string, collection string, logf logger.Logf) Config {
```

Comment:
This is an awkward API for an exported function.

What are you're thoughts on the following:
```go
// LoadConfig reads a JSON file as a Config.
func LoadConfig(file string) (Config, error)

// Store atomically writes the Config as a JSON file.
func (Config) Store(file string) error

// Validate verifies that the Config matches the collection,
// and that the PrivateID and PublicID pair and sensible.
func (Config) Validate(collection string) bool

// Initialize initializes the Config with the collection.
// If the PrivateID is zero, it generates a new one.
func (*Config) Initialize(collection string)
```

Also, I think the ways that we would use this in `logpolicy` and `tsnet` would slightly differ.
`logpolicy` probably does a best effort since we want `tailscaled` to be as autonomous as possible.
It'd be a sad day if upgrading tailscaled prevented it from starting up.

Possible usage in `logpolicy.go`:
```go
c, err := LoadConfig(cfgPath)
if err != nil {
	earlyLogf("logpolicy.LoadConfig %v: %v", cfgPath, err)
}
if !c.Validate(collection) {
	c.Initialize(collection)
	if err := c.Store(cfgPath); err != nil {
		earlyLogf("logpolicy.Config.Store %v: %v", cfgPath, err)
	}
}
```

On the other hand, I feel like `tsnet` adheres to stronger checks since they are usually managed services,
so we would want to eagerly return an error.

Possible usage in `tsnet.go`:
```go
c, err := logpolicy.LoadConfig(cfgPath)
switch {
case err == nil:
	if !c.Validate(logtail.CollectionNode) {
		return fmt.Errorf("logpolicy.Validate failed for %v", cfgPath)
	}
case os.IsNotExist(err):
	c.Initialize(logtail.CollectionNode)
	if err := c.Store(cfgPath); err != nil {
		return fmt.Errorf("logpolicy.Config.Store for %v: %w", cfgPath, err)
	}
default:
	return fmt.Errorf("logpolicy.LoadConfig for %v: %w", cfgPath, err)
}
```

https://go.dev/play/p/is9IXDK367g

Reply:
totally agreed... I didn't *love* this, but really like breaking it apart like you have!  I'll probably pick this back up next week.